File: s3/sdk/ruby/s3.rb
require 'aws-sdk-s3' # AWS SDK for interacting with S3
require 'pry' # Debugging tool (optional, can be removed if not needed)
require 'securerandom' # Library for generating unique identifiers

# Get the bucket name from environment variables
bucket_name = ENV['BUCKET_NAME']
# Define the AWS region where the bucket will be created
region = 'eu-central-1'

# Output the bucket name to the console (for debugging purposes)
puts bucket_name

# Initialize the S3 client
client = Aws::S3::Client.new

# Create a new S3 bucket in the specified region
resp = client.create_bucket({
  bucket: bucket_name,
  create_bucket_configuration: {
    location_constraint: region
  }
})
#binding.pry
# Generate a random number of files (between 1 and 6) to create and upload
number_of_files = 1 + rand(6)
puts "number_of_files: #{number_of_files}"

# Iterate to create and upload each file
number_of_files.times.each do |i|
  puts "i: #{i}" # Output the current iteration index

  # Define the filename and local file path
  filename = "file_#{i}.txt"
  output_path = "/tmp/#{filename}"

  # Open the file for writing and write a unique identifier to it
  File.open(output_path, "w") do |f|
    f.write SecureRandom.uuid
  end

  # Open the file in binary read mode and upload it to the S3 bucket
  File.open(output_path, 'rb') do |file|
    client.put_object(
      bucket: bucket_name, # The name of the S3 bucket
      key: filename, # The key (name) of the object in the bucket
      body: file # The file content to upload
    )
  end
end

File: s3/sdk/php/s3.php
<?php
require __DIR__ . '/vendor/autoload.php';

use Aws\S3\S3Client;

class S3Example
{

    /**
     * @var S3Client
     */
    private $s3client;
    /**
     * @var string
     */
    private $bucketName;

    public function createClient(): void
    {
        $region = 'eu-central-1';

        $this->s3client = new S3Client([
            'region' => $region,
        ]);
    }

    public function createBucket($name): void
    {
        $this->bucketName = $name;
        if (!$this->s3client) {
            $this->createClient();
        }

        try {
            $this->s3client->createBucket([
                'Bucket' => $this->bucketName
            ]);
            echo "Created bucket named: $this->bucketName \n";
        } catch (Exception $exception) {
            echo "Failed to create bucket $this->bucketName with error: " . $exception->getMessage();
            exit("Please fix error with bucket creation before continuing.");
        }
    }

    /**
     * @return void
     */
    public function putObjectToBucket(): void
    {
        $numberOfFiles = rand(1, 6);
        for ($i = 0; $i < $numberOfFiles; $i++) {
            $fileName = "file_$i.txt";
            $outputPath = "/tmp/$fileName";
            file_put_contents($outputPath, uniqid());
            try {
                $this->s3client->putObject([
                    'Bucket' => $this->bucketName,
                    'Key' => $fileName,
                    'SourceFile' => $outputPath
                ]);
                echo "Uploaded $fileName to $this->bucketName.\n";
            } catch (Exception $exception) {
                echo "Failed to upload $fileName with error: " . $exception->getMessage();
                exit("Please fix error with file upload before continuing.");
            }
        }
    }
}

$s3Example = new S3Example();

$bucketName = $argv[1] ?? null;
if ($bucketName) {
    $s3Example->createBucket($bucketName);
    $s3Example->putObjectToBucket();
} else {
    echo "Put Bucket Name as a first argument.\n";
}

